boxsdk/object/metadata.py

```diff
@@ -206,6 +207,33 @@ def create(self, metadata):
             headers={b'Content-Type': b'application/json'},
         ).json()
 
+    @api_call
+    def set(self, metadata):
+        """
+        Set the metadata instance on a :class:`Folder` or :class:`File`. Attempts to first create metadata on a
+        :class:`Folder` or :class:`File`. If metadata already exists then attempt an update.
+
+        :param metadata:
+            The key/value pairs to be stored in this metadata instance on Box.
+        :type metadata:
+            `dict`
+        :return:
+            A dictionary containing the key/value pairs for this metadata object.
+        :rtype:
+            :class:`Metadata`
+        """
+        metadata_value = None
+        try:
+            metadata_value = self.create(metadata)
+        except BoxAPIException as err:
+            if err.status == 409:
+                metadata_instance = Metadata(self._session, self, self._scope, self._template)
```

Comment:
Isn't this object (`self`) already a `Metadata` instance?  Could you just call `self.start_update()` instead?